csharp-basics/exercises/Polymorphism/VendingMachine/VendingMachine.cs

```diff
@@ -12,24 +12,24 @@ public class VendingMachine : IVendingMachine
         public bool HasProducts { get; set; }
         public Money Amount { get; set; }
         public Product[] Products { get; set; }
-        public VendingMachine()
-        {
-            this.Manufacturer = null;
-            this.HasProducts = false;
-            this.Products = null;
-        }
 
         public VendingMachine(string manufacturer, bool hasProducts, Money amount, Product[] products)
         {
-            this.Manufacturer = manufacturer;
-            this.HasProducts = hasProducts;
-            this.Amount = amount;
-            this.Products = products;
+            Manufacturer = manufacturer;
+            HasProducts = hasProducts;
+            Amount = amount;
+            Products = products;
         }
 
         public Money InsertCoin(Money amount)
         {
-            return Amount;
+            if (amount.Euros == 0 && amount.Cents == 10 || amount.Euros == 0 && amount.Cents == 20 ||
```

Comment:
pieņemamās naudas vienības var sadefinēt kkādā listā un pēc tam vienkārši pārbaudīt vai lists satur konkrēto vienību. :) 
ja satur tad turpināt. ja ne tad izlaist. :) un tas ifs uzreiz būs pārlasāmāks un saprotamāks. :)